commit cbb3b37ec7bc1e3d35b458d67b724883b9ff15fb
Author: Gilles Dartiguelongue <eva@gentoo.org>
Date:   Mon Sep 28 01:55:39 2020 +0200

    Make gst-inspect calls optional at build-time
    
    Crashes on Gentoo due to plugins trying to access dri nodes which are sandboxed

diff --git a/meson_options.txt b/meson_options.txt
index 7070fd5d0..678b8f588 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -3,3 +3,4 @@ option('enable-python', type: 'combo', choices: ['yes', 'no', 'auto'], value: 'a
 option('with-plugins', type: 'combo', choices: ['all', 'none', 'auto'], value: 'auto', description: 'Which Totem plugins to compile (default: auto; "all", "none" and "auto" are valid)')
 option('enable-gtk-doc', type: 'boolean', value: false, description: 'use gtk-doc to build documentation')
 option('profile', type: 'combo', choices: ['default', 'development'], value: 'default', description: 'Build profiles')
+option('gst-inspect', type: 'boolean', value: true, description: 'build-time test of available gstreamer plugins')
diff --git a/src/backend/meson.build b/src/backend/meson.build
index fbf74258b..1e025a5b9 100644
--- a/src/backend/meson.build
+++ b/src/backend/meson.build
@@ -1,5 +1,6 @@
 backend_inc = include_directories('.')
 
+if get_option('gst-inspect')
 gst_inspect = find_program(
   'gst-inspect-1.0',
   join_paths(gst_dep.get_pkgconfig_variable('toolsdir'), 'gst-inspect-1.0'),
@@ -31,6 +32,7 @@ foreach plugin: gst_good_plugins
   assert(r.returncode() == 0,
          'Cannot find required GStreamer-1.0 plugin "' + plugin + '". It should be part of gst-plugins-good. Please install it.')
 endforeach
+endif
 
 sources = files(
   'bacon-time-label.c',
